fastembed/common/model_management.py

```
@@ -281,6 +281,16 @@ def _save_file_metadata(model_dir: Path, meta: dict[str, dict[str, int | str]])
 
         return result
 
+    @staticmethod
+    def _safe_extractall(tar: tarfile.TarFile, cache_dir: str) -> None:
+        # Manual traversal guard for interpreters without the 'data' extraction filter.
+        base = os.path.realpath(cache_dir)
+        for member in tar.getmembers():
+            target = os.path.realpath(os.path.join(cache_dir, member.name))
+            if os.path.commonpath([base, target]) != base:
+                raise ValueError(f"Blocked path traversal in tar member: {member.name!r}")
+        tar.extractall(path=cache_dir)
```

Ruff (0.16.1)

[error] Uses of `tarfile.extractall()`

(S202)